### DISCUSSION TITLE
Proposal: dot equals assignment operator

### DISCUSSION THREAD

**Discussion by BanForFun** [dotnet/csharplang#3051] [Language Ideas]
## Summary

Consider for example that we get some input from the user and need to truncate it to 10 characters, and use the `Substring` method to do so.

```c#
string userInput = Console.ReadLine();
//Do stuff with full string...

userInput = userInput.Substring(0, 10);
```

My suggestion is the addition of a `.=` operator, with the last line being written as: 
`userInput .= Substring(0, 10);`

## Detailed design

This operator could be used with instance methods (as shown above), extension methods, properties and fields. Of course, only if the types are compatible.

With this assignment operator could come an equality one as well (`.==`). For example to check if a string is all uppercase:

````c#
string checkCase = "abc";
bool isUpper = checkCase == checkCase.ToUpper();
````

We would instead use: `bool isUpper = checkCase .== ToUpper()`

## Edit: Alternative syntax
As currently there are no equality operators with 3 characters, here is an alternative syntax which would potentially easier to adapt to: `== .`
The second example could then be `bool isUpper = checkCase == .ToUpper()`

The same goes for the assignment operator which could be: `= .`
The first example would then be `userInput = .Substring(0, 10)`

## Alternatives

For short variable names the traditional way isn't such a problem, but for longer ones this would (in my opinion) be usefull.

**Comment by AlgorithmsAreCool** [comment]
While I've certianly written code in that pattern, i can't imagine it is done often enough to justify a new operator. 

Every new feature is another things for all C# devs to have to learn and adjust to, i don't think this pays for itself.

**Comment by svick** [comment]
The assignment part of this is a duplicate of https://github.com/dotnet/csharplang/discussions/500.
